Proposal: Use `prettier` to format front-end code

dogmar opened the pull request:
<!--- Hello Plural contributor! It's great to have you on board! -->

## Summary
<!-- Describe your changes here, include the motivation/context, test coverage, -->
<!-- the type of change i.e. breaking change, new feature, or bug fix -->
<!-- and related GitHub issue or screenshots (if applicable). -->

<!-- Adding a meaningful title and description allows us to better communicate -->
<!-- your work with our users. -->

## Labels
<!-- For breaking changes, add the `breaking-change` label.️ -->
<!-- For bug fixes, add the `bug-fix` label. -->
<!-- For new features and notable changes, add the `enhancement` label. -->


## Test Plan
<!--- Please describe the tests you have added and your testing environment (if applicable). -->


## Checklist
<!--- Go over all the following points to make sure you've checked all that apply before merging. -->
<!--- If you're unsure about any of these, don't hesitate to ask in our Discord. -->

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

maciaszczykm commented on assets/package.json:
```diff
@@ -11,15 +11,18 @@
+    "graphql:codegen": "graphql-codegen --config codegen.yml",
+    "prettier-write": "yarn prettier --write ./src",
+    "prettier-check": "yarn prettier --check ./src",
+    "postinstall": "cd .. && husky install assets/.husky"
```
We can run it automatically like [here](https://github.com/kubernetes/dashboard/blob/master/modules/web/.yarnrc.yml#L9) with [postinstall](https://github.com/kubernetes/dashboard/blob/master/modules/web/.yarn/plugins/%40yarnpkg/plugin-postinstall.cjs) plugin.

dogmar replied:
Oh great! I was struggling to find a way to get husky installed automatically with yarn. I'll take a look at this plugin.

maciaszczykm commented on assets/src/components/account/groups/Group.tsx:
```diff
@@ -30,29 +34,21 @@ export default function Group({ group, q }: { group: GroupT; q: any }) {
-        gap="large"
-        align="center"
-      >
+      <Flex flex={false} direction="row" gap="large" align="center">
```
One thing to consider is keeping args in separate lines. I think it was easier to read. What do you think?

floreks replied:
Agree, I also prefer arg-per-line style.

dogmar replied:
Also makes for cleaner diffs. Looks like we all agree here!

floreks commented:
Great idea in general. We should use an external formatter to avoid colliding with each other's IDEs.

maciaszczykm commented:
@dogmar My recent changes in master branch have caused a lot of conflicts in this pull request, so I have resolved them. I have also added configuration files for prettier with `singleAttributePerLine: true`. This is just a proposal, let me know what you think and feel free to revert any changes that I have made here.

dogmar commented:
I missed that `singleAttributePerLine: true` was even an option. I prefer it as well. However, I think we should stick with plural's default for `printWidth`. They have an explanation [here](https://prettier.io/docs/en/options.html#print-width). If these rest of you still feel strongly about increasing it, I could be convinced to go to `100`.

maciaszczykm commented:
@dogmar We can keep 80 for now, and go back to the topic if there will be a need. It is not a strong opinion from my side, rather just me being used to it. Looking at the lines of code stats for this PR (+23k, -51k) the balance is pretty good anyways.

dogmar commented:
I went ahead and changed this PR so all the commits until the very end leave `assets/src` alone, then rebased to be sure we had the very latest code from `origin/main` before running `yarn fix`. Also added the postinstall yarn plugin to install husky (thanks, @maciaszczykm !). Should be ready for final review.

maciaszczykm commented on assets/package.json:
```diff
@@ -11,15 +11,19 @@
-    "graphql:codegen": "graphql-codegen --config codegen.yml"
+    "graphql:codegen": "graphql-codegen --config codegen.yml",
+    "prettier-write": "yarn prettier --write ./src",
+    "prettier-check": "yarn prettier --check ./src"
```
Can you move it to lint:format above?

maciaszczykm replied:
Perhaps we should also rename lint to check?

dogmar replied:
Removed those. I think leaving everything as `lint` is fine and then we don't have to worry about breaking CI.

maciaszczykm commented on assets/package.json:
```diff
@@ -139,19 +143,28 @@
+    "husky": "8.0.3",
     "jsdom": "21.1.0",
+    "lint-staged": "13.2.0",
+    "npm-run-all": "^4.1.5",
```
I forgot to remove ^.

dogmar replied:
fixed

maciaszczykm reviewed (approved):
LGTM, with some tiny nit-picks.
